### PR TITLE
17765 complete resolution fix

### PR DESCRIPTION
**Pull request by JazzarKarim**
*Issue #:* /bcgov/entity#17765

*Description of changes:*
- Fixed the unit tests of `CompleteResolution.vue` after the latest hotfix has been deployed
- Moved from happy-dom to jsdom
- Fixed other tests
- Removed isVitestRunning flag

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-565-0n72txxt.web.app

File: src/components/Dissolution/CompleteResolution.vue [hunk 1]
```diff
@@ -574,7 +574,6 @@ export default class CompleteResolution extends Mixins(CommonMixin, DateMixin) {
     // Note: the image file path did not resolve correctly when using the require function directly.  In order
     // to get the image path resolving correctly, needed to get the image context first and use that to build
     // the final image file path
-    if (this.isVitestRunning) return ''
```
Comment (JazzarKarim) [inline src/components/Dissolution/CompleteResolution.vue:577]: Travis expressed his concerns with this flag. I removed it from here and elsewhere. Unit test was failing:
invalid url

After doing some research (https://github.com/mswjs/msw/discussions/1521), I was able to make the unit tests run by moving from happy-dom to jsdom.

File: tests/unit/AccountContactMissingDialog.spec.ts [hunk 2]
```diff
@@ -38,19 +38,16 @@ describe('AccountContactMissingDialog.vue', () => {
       propsData: { dialog: true }
     })
 
-    // Override window.location.href
-    let redirectedUrl = ''
-    // eslint-disable-next-line accessor-pairs
-    Object.defineProperty(window.location, 'href', {
-      set (url: string) {
-        redirectedUrl = url
```
Comment (JazzarKarim) [inline tests/unit/AccountContactMissingDialog.spec.ts:46]: This test was failing after moving from happy-dom to jsdom:
cannot redefine property

The new code fixes it. Reference: https://github.com/jestjs/jest/issues/5124

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-565-0n72txxt.web.app

File: src/components/Dissolution/CompleteResolution.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/Dissolution/CompleteResolution.vue:577]: How about changing to `jsdom` as needed? In theory, `happy-dom` is faster or  better or something (happier?).

File: tests/unit/AccountContactMissingDialog.spec.ts (hunk 2 above)
Comment (severinbeauvais) [reply tests/unit/AccountContactMissingDialog.spec.ts:46]: Nice find!

**Review by severinbeauvais — APPROVED** [review]

File: src/components/Dissolution/CompleteResolution.vue (hunk 1 above)
Comment (JazzarKarim) [reply src/components/Dissolution/CompleteResolution.vue:577]: Oh, you mean leave `jsdom` here and keep `happy-dom` in the other UIs? Sure! That works too 🙌

Comment (severinbeauvais) [reply src/components/Dissolution/CompleteResolution.vue:577]: Yes, leave the others until we have a reason to update them.

Comment (JazzarKarim) [reply src/components/Dissolution/CompleteResolution.vue:577]: All right, sounds good.

**Review by seeker25 — APPROVED** [review]

File: src/components/common/Actions.vue [hunk 3]
```diff
@@ -286,8 +285,8 @@ export default class Actions extends Mixins(CommonMixin, DateMixin, FilingTempla
         this.setIsFilingPaying(false)
       }
     } else {
-      // don't call window.scrollTo during Vitest tests because happy-dom doesn't implement it
-      if (!this.isVitestRunning) window.scrollTo({ top: 0, behavior: 'smooth' })
+      // don't call window.scrollTo during Vitest tests because jsdom doesn't implement it
```
Comment (severinbeauvais) [inline src/components/common/Actions.vue:288]: Hehe, obsolete comment.

**Pull request merged**